mesonbuild/interpreter.py

```diff
@@ -2830,6 +2867,29 @@ def func_message(self, node, args, kwargs):
     def message_impl(self, argstr):
         mlog.log(mlog.bold('Message:'), argstr)
 
+    @noArgsFlattening
+    @noKwargs
+    @FeatureNew('summary', '0.53.0')
+    def func_summary(self, node, args, kwargs):
+        if len(args) != 2:
+            raise InterpreterException('Summary accepts exactly two arguments.')
+        section, values = args
+        if not isinstance(section, str):
+            raise InterpreterException('Argument 1 must be a string.')
+        if not isinstance(values, dict):
+            raise InterpreterException('Argument 2 must be a dictionary.')
+        if self.subproject not in self.summary:
+            self.summary[self.subproject] = Summary(self.active_projectname, self.project_version)
+        self.summary[self.subproject].add_section(section, values)
+
+    def _print_summary(self):
+        mlog.log('')  # newline
+        main_summary = self.summary.pop('')
+        for project, summary in sorted(self.summary.items()):
```

Comment:
__[Flake8]__ Loop control variable 'project' not used within the loop body. If this is intended, start the name with an underscore. ([see on Sider](https://sider.review/gh/repos/19784232/pulls/4649#issue-4934987))

<details>

| Rule |
| - |
| B007 |

</details>
<sub>posted by Sider</sub>